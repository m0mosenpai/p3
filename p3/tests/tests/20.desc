Truncate history when it is reduced in size. E.g. history set 10, (10 unique commands), history set 5, history set 10. There should only be 5 items in history.
