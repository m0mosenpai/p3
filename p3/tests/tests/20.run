../solution/wsh <tests/20.wsh
