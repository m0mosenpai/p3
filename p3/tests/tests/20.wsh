history set 10
echo 1
echo 2
echo 3
echo 4
echo 5
echo 6
echo 7
echo 8
echo 9
echo 10
history
history set 5
history
history set 10
history
