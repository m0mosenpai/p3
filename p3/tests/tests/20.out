wsh> wsh> 1
wsh> 2
wsh> 3
wsh> 4
wsh> 5
wsh> 6
wsh> 7
wsh> 8
wsh> 9
wsh> 10
wsh> 1) echo 10
2) echo 9
3) echo 8
4) echo 7
5) echo 6
6) echo 5
7) echo 4
8) echo 3
9) echo 2
10) echo 1
wsh> wsh> 1) echo 10
2) echo 9
3) echo 8
4) echo 7
5) echo 6
wsh> wsh> 1) echo 10
2) echo 9
3) echo 8
4) echo 7
5) echo 6
wsh> 